### PR TITLE
Update gate Jenkins endpoint

### PR DESCRIPTION
**Pull request by AMeng**
Fixes: https://github.com/spinnaker/spinnaker/issues/605
Depends on: https://github.com/spinnaker/gate/pull/161

### CODE REVIEW

File: app/scripts/modules/core/ci/jenkins/igor.service.js [hunk 1]
```diff
@@ -9,19 +9,19 @@ module.exports = angular.module('spinnaker.core.ci.jenkins.igor.service', [
...
     }
 
     function listJobsForMaster(master) {
-      return Restangular.one('builds', master).all('jobs').getList();
+      return Restangular.one('v2').one('builds', master).all('jobs').getList();
     }
 
     function listBuildsForJob(master, job) {
-      return Restangular.one('builds', master).one('jobs', job).all('builds').getList();
+      return Restangular.one('v2').one('builds', master).one('builds').one('jobs', job).getList();
```
Comment (tomaslin) [inline app/scripts/modules/core/ci/jenkins/igor.service.js]: this url seems to be incorrect, it adds an additional /jobs/ 

http://192.168.99.100:8084/v2/builds/spinnaker/builds/jobs/SPINNAKER-package-igor

http://192.168.99.100:8084/v2/builds/spinnaker/builds/SPINNAKER-package-igor

**Comment by riltsken** [comment]
@AMeng To not encode the URI component you'll need to do this https://github.com/riltsken/deck/commit/7900ccfd207b226070096b11515e4400fc51d787 I can't add commits to this branch unfortunately since it's under your account.

File: app/scripts/modules/core/ci/jenkins/igor.service.js [hunk 2]
```diff
@@ -9,19 +9,24 @@ module.exports = angular.module('spinnaker.core.ci.jenkins.igor.service', [
...
     }
 
     function listJobsForMaster(master) {
-      return Restangular.one('builds', master).all('jobs').getList();
+      return Restangular.one('v2').one('builds', master).all('jobs').getList();
     }
 
     function listBuildsForJob(master, job) {
-      return Restangular.one('builds', master).one('jobs', job).all('builds').getList();
+      return Restangular.one('v2').one('builds', master).one('builds').one(job).getList();
```
Comment (AMeng) [inline app/scripts/modules/core/ci/jenkins/igor.service.js]: Need to make sure job is not encoded here either

**Comment by anotherchrisberry** [comment]
LGTM - needs a rebase

**Comment by AMeng** [comment]
Rebased. Remember that this depends on spinnaker/gate#161. It will break the UI without it.

**Pull request merged**